### ISSUE TITLE
Utilize WP img alt attribute for gallery images

### ISSUE THREAD

**Issue by coreypotter** [netrivet/prophoto-issues#737]
At this time the pp galleries don't specify an alt attribute in the code. I would love to see the WordPress alt text used here. I've had several people in my group wishing for this.

**Comment by benjamintennant** [comment]
+1 Matt Thompson ticket 162018

**Comment by brianium** [comment]
Any images loaded at page load should have their alt attributes filled from the WordPress alt text (correct me if that is not the case). Where we don't add alt text is on images that are lazily fetched via ajax - i.e images loaded as you page through a large gallery.

It should not be terribly difficult to add this alt text, but what would be the benefit? It seems unlikely to aid search results at that point, but maybe provides an accessibility win?

**Comment by coreypotter** [comment]
@brianium from what I'm seeing, the alt text isn't used at page load or when paging through the gallery (just double checked on a masonry gallery). 

For search, I'd argue that the alt text (when used correctly) makes a big difference in adding context, especially for a gallery page with little text. I would say that many people don't use it correctly, and I would also say that in the grand scheme of things, it is a small factor for most users. However, there are some use cases where I may want to insert a gallery onto a page where the alt text would carry heavier weight for a search engine. Lets say for example I'm using a gallery on a blog post, and only inserting 5-10 images. The extra context and information in the alt text can add certainty and help images and the entire page rank, especially for more specific keywords.

Not to mention, including an alt attribute for every image is pretty standard as far as code quality goes. 
https://support.google.com/webmasters/answer/114016?hl=en
https://www.w3.org/QA/Tips/altAttribute

**Comment by brianium** [comment]
via @vkozintsev:

There is no alt tags for images when we create a gallery as the thumbnail grid. I would love to see the Media alt text used here.
![alt_tag](https://cloud.githubusercontent.com/assets/26632225/24261841/ae46248e-0fce-11e7-8f30-936754cbbef8.png)

**Comment by brianium** [comment]
@coreypotter - I agree with you completely on the benefits of alt attributes :) - We try to include them on any images that are not inserted via JavaScript.

Thumbnail galleries are an interesting exception to how most galleries are rendered. Thumbnail galleries are entirely inserted via JavaScript - that is both the grid and the gallery that is spawned when clicking on an individual thumb are dynamically inserted.

If you were to create a slider gallery, or a carousel gallery (coming in 6.18), those gallery images are rendered on the server - and do have alt attributes.

The question becomes: Is there a benefit to adding alt attributes on images that are dynamically inserted via JavaScript? The answer very well could be yes - I'm not sure :) - Most of the stuff I have read has not give me a ton of clarity on what screen readers/bots are capable of in terms of JavaScript.

**Comment by benjamintennant** [comment]
+1 Jackson Eddy ticket 166362 - wants alt text on thumbnail images in thumbnail style gallery

**Comment by postmastersteve** [comment]
This was completed in https://github.com/netrivet/prophoto-issues/issues/737
